### PR TITLE
Update Vesting concept

### PR DESCRIPTION
**Pull request by csigma-nischay**
Previous concept : Vesting is started after the cliff ends.
Updated: Vesting will start when cliff begins or from the Vesting schedule start time

### CODE REVIEW

File: test/TokenVesting.js [hunk 1]
```diff
@@ -436,6 +415,78 @@ describe("TokenVesting", function () {
...
+      expect(await tokenVesting.computeReleasableAmount(vestingScheduleId)).to.be.equal(246);
+
+      // Date -May 01, 2024 
+      // Return - 246
+      await tokenVesting.setCurrentTime(baseTime + cliff+ 2592000);
+      expect(await tokenVesting.computeReleasableAmount(vestingScheduleId)).to.be.equal(328);
+
+
+      // set time to halfway through the vesting period (12 months)
+      const halfwayThrough = baseTime + (cliff + duration)/ 2;
```
Comment (dhavaldavra) [inline test/TokenVesting.js:475]: In our case, halfwayThrough = baseTime + (duration / 2)

and expectation is the claimable amount should be grater than 500.

File: test/TokenVesting.js [hunk 2]
```diff
@@ -436,6 +415,78 @@ describe("TokenVesting", function () {
...
+      await tokenVesting.setCurrentTime(justBeforeCliff);
+      expect(await tokenVesting.computeReleasableAmount(vestingScheduleId)).to.be.equal(0);
+
+      // Date -Apr 01, 2024 
+      // Return - 246
+      await tokenVesting.setCurrentTime(baseTime + cliff);
+      expect(await tokenVesting.computeReleasableAmount(vestingScheduleId)).to.be.equal(246);
+
+      // Date -May 01, 2024 
+      // Return - 246
```
Comment (dhavaldavra) [inline test/TokenVesting.js:469]: Return value is not correct

File: test/TokenVesting.js [hunk 3]
```diff
@@ -436,6 +415,78 @@ describe("TokenVesting", function () {
...
+
+      // set time to halfway through the vesting period (12 months)
+      const halfwayThrough = baseTime + (cliff + duration)/ 2;
+      await tokenVesting.setCurrentTime(halfwayThrough);
+
+      // check that vested amount is greater than 0
+      expect(await tokenVesting.computeReleasableAmount(vestingScheduleId)).to.be.gt(400);
+
+      // set time to the end of the vesting period (12 months + 1 day)
+      const endOfVesting = baseTime + 316322400;
```
Comment (dhavaldavra) [inline test/TokenVesting.js:482]: 366 days = 31,622,400 seconds

**Review by dhavaldavra — CHANGES_REQUESTED** [review]
Please make some corrections in test cases

File: test/TokenVesting.js (hunk 3 above)
Comment (csigma-nischay) [reply test/TokenVesting.js:482]: its for leap year

**Review by dhavaldavra — APPROVED** [review]
Looks good

**Pull request merged**